OpenRA.Mods.Common/AI/States/GroundStates.cs

```diff
@@ -140,9 +149,14 @@ public void Tick(Squad owner)
 				}
 			}
 
-			foreach (var a in owner.Units)
-				if (!BusyAttack(a))
-					owner.Bot.QueueOrder(new Order("Attack", a, false) { TargetActor = owner.Bot.FindClosestEnemy(a.CenterPosition) });
+			// Switch target during fight
+			var notBusyUnits = owner.Units.Where(a => !BusyAttack(a));
```

Comment:
This means you'll need to enumerate the units twice and check if they are busy attacking each time. I would suggest using the null-check I suggested to avoid this overhead.